fix: NaN sum if evict all and add back element

arthhhhh23 opened the pull request:
When `Evict` removes the last element from the `Window`, it sets `w.sum` to `math.NaN`.
This in turn makes counter to stuck at `NaN` forever, as increment of `w.sum += value` always yields `NaN`.

This PR changes `Evict` logic to set `w.sum = 0` when all values are evicted (this is consistent with default state of empty `Window` created by `NewWindow`) and avoids the `NaN` issue altogether.

arthhhhh23 committed:
fix: NaN sum if evict all and add back element

C-Pro commented on rolling.go:
```diff
@@ -110,7 +110,12 @@ func (w *Window) AddAt(value float64, at time.Time) {
 	w.cnt++
-	w.sum += value
+
+	if w.cnt == 1 {
```
Maybe we should set sum to 0 on evict instead?
I think it makes more sense than NaN since default value is 0 when new empty window is created

arthhhhh23 replied:
I thought you wanted it to be NaN since no element means there nothing to sum up

arthhhhh23 replied:
but in practical terms, it might be better to use 0 directly like you said. I'll go with your preferred approach

C-Pro replied:
> I thought you wanted it to be NaN since no element means there nothing to sum up

That was the idea, but as you found out it does not work nicely. And it is inconsistent with the initial empty window state that is created in New.

arthhhhh23 replied:
updated

arthhhhh23 committed:
refactor: make sum=0 if count==0 and revert AddAt change

C-Pro committed:
Remove osv-scanner deprecated option

`osv-scanner` deprecated `--skip-git` and returns an error when provided, breaking backwards compatibility 🤷

https://github.com/lablup/backend.ai/pull/4205

C-Pro committed:
Ignore GO-2025-3750

os.OpenFile is not used
